lib/contracts/contributionreward.ts

```diff
@@ -157,6 +157,213 @@ export class ContributionRewardWrapper extends ExtendTruffleContract {
+
+    const defaults = {
+      avatar: undefined,
+      proposalId: undefined,
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    if (!options.avatar) {
+      throw new Error("avatar address is not defined");
+    }
+
+    const tx = await this.contract.redeemNativeToken(
+      options.proposalId,
+      options.avatar,
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  /**
+   * Redeem ether reward for the beneficiary of the proposal
+   * @param {ContributionRewardSpecifiedRedemptionParams} opts
+   */
+  public async redeemEther(opts = {}): Promise<ArcTransactionResult> {
+
+    const defaults = {
+      avatar: undefined,
+      proposalId: undefined,
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    if (!options.avatar) {
+      throw new Error("avatar address is not defined");
+    }
+
+    const tx = await this.contract.redeemEther(
+      options.proposalId,
+      options.avatar,
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  /**
+   * Return all proposals ever created under the given avatar.
+   * Filter by the optional proposalId.
+   * @param opts
+   */
+  public async getDaoProposals(opts = {}): Promise<Array<ContributionProposal>> {
+
+    const defaults: GetDaoProposalsParams = {
+      avatar: undefined,
+      proposalId: null,
+    };
+
+    const options: GetDaoProposalsParams = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.avatar) {
+      throw new Error("avatar address is not defined");
+    }
+
+    const proposals = new Array<ContributionProposal>();
+
+    if (options.proposalId) {
+      const orgProposal = await this.contract.organizationsProposals(options.avatar, options.proposalId);
+      const proposal = this.orgProposalToContributionProposal(orgProposal, options.proposalId);
+      proposals.push(proposal);
+    } else {
+      const eventFetcher = this.contract.NewContributionProposal({ _avatar: options.avatar }, { fromBlock: 0 });
+      await new Promise((resolve) => {
+        eventFetcher.get(async (err, events) => {
+          for (const event of events) {
+            const proposalId = event.args._proposalId;
+            const orgProposal = await this.contract.organizationsProposals(options.avatar, proposalId);
+            const proposal = this.orgProposalToContributionProposal(orgProposal, proposalId);
+            proposals.push(proposal);
+          }
+          resolve();
+        });
+      });
+    }
+
+    return proposals;
+  }
+
+  /**
+   * Return a list of executed (passed) proposals that have rewards
+   * waiting to be redeemed by the given beneficiary.
+   * @param opts
+   */
+  public async getBeneficiaryRewards(opts = {}): Promise<Array<ProposalRewards>> {
```

Comment:
maybe should say "unredeemed" in the name?

Reply:
OK, I took the liberty of keeping the name but having it also return the total amounts redeemable.